src/main/java/io/supertokens/storage/mongodb/Queries.java

```diff
@@ -151,6 +158,87 @@ static KeyValueInfoWithLastUpdated getKeyValue_Transaction(Start start, String k
         return KeyValueInfoLastUpdatedRowMapper.getInstance().mapOrThrow(result);
     }
 
+    static void deleteKeyValue(Start start, String key) {
+        MongoDatabase client = ConnectionPool.getClientConnectedToDatabase(start);
+        MongoCollection collection = client.getCollection(Config.getConfig(start).getKeyValueCollection());
+
+        collection.deleteOne(Filters.eq("_id", key));
+    }
+    
+    static List<KeyValueInfo> getArrayKeyValue_Transaction(Start start, String key) throws StorageQueryException{
+        MongoDatabase client = ConnectionPool.getClientConnectedToDatabase(start);
+        MongoCollection collection = client.getCollection(Config.getConfig(start).getKeyValueCollection());
+        Document result = (Document) collection.find(Filters.eq("_id", key)).first();
+        if (result == null) {
+            return null;
```

Comment:
Ideally, you should return an empty list here? Since there are no keys? This will then be similar semantic to sql dbs, in which if there are no keys, it returns an empty list. I know that you have a `null` check in the calling function, but I prefer if it's here - lesser changes of an error if this function is used elsewhere in the future.